dds/DCPS/transport/framework/TransportRegistry.h

```diff
@@ -125,6 +126,15 @@ class OpenDDS_Dcps_Export TransportRegistry {
 
   OPENDDS_STRING get_config_instance_name(DDS::DomainId_t id);
 
+  bool create_new_transport_instance_for_participant(DDS::DomainId_t id, OPENDDS_STRING& config_name);
+
+  // The TransportRegistry needs to know the domains for a given
+  // config in order to know when to create new transport instances
+  // for separate participants in the same domain. This is called
+  // from the Service_Particpant since it knows if the GlobalTransportConfig
+  // has been set or overriden by a command line option.
+  void associate_domain_to_config(DDS::DomainId_t id, OPENDDS_STRING cfg);
```

Comment:
```suggestion
  void associate_domain_to_config(DDS::DomainId_t id, const OPENDDS_STRING& cfg);
```
And maybe switch to TString?